### ISSUE TITLE
Test Dagger on Windows

### ISSUE THREAD

**Issue by grouville** [dagger/dagger#3462]
Tests were broken on Mac #3433 and #3437, due to Path/Symlink differences between OS and versions.

Related to #3012
Reminder to check that the current version of Dagger works on Windows.

**Comment by sipsma** [comment]
This issue is coming to a greater head w/ the engine image and local binaries. We need to explicitly choose whether or not we want to package helper windows binaries (or `dagger-engine-session` as it will be renamed soon) inside the engine image.

If we don't, dagger will not be usable on Windows at all to start (whereas previously it might have been sort of usable even if it was completely untested).

The main limiting factor in this is that we need to test it on Windows. I personally don't have a windows machine, so I can't even do manual testing without setting up a VM and such.

cc @aluzzardi @shykes something to be aware of for the upcoming release.

**Comment by shykes** [comment]
I think we should. It makes sense at least on paper. I can't speak to difficulties of testing & packaging, which may lead to implementation tradeoffs.

**Comment by shykes** [comment]
by the way, something just occurred to me. In the future, we could add to the engine API itself the ability to send the correct helper to the client. so, instead of having to pull the image itself, run it, and extracting the binary via docker, the client could instead connect to the engine, and then simply download the correct helper.

**Comment by mircubed** [comment]
@samalba  @shykes since this didn't make it as part of the python release - who should we assign this to, so we can make sure it is completed?

**Comment by jedevc** [comment]
cc @matipan @gerhard - found this one while looking through some of our old issues.

Once the migration to the new workers is complete, it would be really nice to have some windows ci workers as well - potentially happy to pick that up myself, not sure if we're tracking it somewhere in linear already?
